### PR TITLE
feat(teams): allow editing team membership on own member details page

### PR DESCRIPTION
**Pull request by MaxBittker**

### CODE REVIEW

File: src/sentry/api/serializers/models/organization_member.py [hunk 1]
```diff
@@ -53,8 +53,9 @@ def get_attrs(self, item_list, user):
 
         # results is a map of member id -> team_slug[]
         for m in member_team_map:
-            results[m['organizationmember_id']].append(
-                teams[m['team_id']].slug)
+            if teams[m['team_id']].status is TeamStatus.VISIBLE:
```
Comment (MaxBittker) [inline src/sentry/api/serializers/models/organization_member.py]: deleted teams were showing up in this list and causing problems

Comment (mattrobenolt) [reply src/sentry/api/serializers/models/organization_member.py]: Just filter this out of the query in the first place.

```python
teams = {team.id: team for team in Team.objects.filter(
    status=TeamStatus.VISIBLE,
    id__in=[item['team_id'] for item in member_team_map])}
```

Or even inside the `OrganizationMemberTeam` query with `team__status=TeamStatus.VISIBLE`

File: src/sentry/api/endpoints/organization_member_details.py [hunk 2]
```diff
@@ -204,6 +199,12 @@ def put(self, request, organization, member_id):
             if not result['role'] in {r.id for r in allowed_roles}:
                 return Response(
                     {'role': 'You do not have permission to invite that role.'}, status=403)
+                # You can't edit your own role
```
Comment (macqueen) [inline src/sentry/api/endpoints/organization_member_details.py]: nit, it'd move this to be right above the block it's describing and fix the indentation

**Review by macqueen — APPROVED** [review]
if you just fix matt's suggestion, lgtm

**Pull request merged**